network_plugin/__init__.py

```diff
@@ -133,23 +136,23 @@ def get_network_name(properties):
 
 
 def is_network_exists(vca_client, network_name):
-    networks = vca_client.get_networks(get_vcloud_config()['vdc'])
-    return any([network_name == net.get_name() for net in networks])
+    return bool(vca_client.get_network(get_vcloud_config()['vdc'], network_name))
+
+
+def is_network_routed(vca_client, network_name):
```

Comment:
We also need to check if network is routed through a particular gateway. Because if it's connected to one network, and we apply our configurations (NAT, FW) on the other - it's not going to work.